code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -313,85 +313,36 @@ emp_act
 	return 1*/
 
 //this proc handles being hit by a thrown atom
-/mob/living/carbon/human/hitby(atom/movable/AM as mob|obj,var/speed = 5)
+/mob/living/carbon/human/hitby(atom/movable/AM, skipcatch = 0, hitpush = 1, blocked = 0)
+	var/obj/item/I
+	var/throwpower = 30
 	if(istype(AM, /obj/item))
-		var/obj/item/I = AM
-
-		if(in_throw_mode && !get_active_hand() && speed <= 5)	//empty active hand and we're in throw mode
-			if(canmove && !restrained())
-				if(isturf(I.loc))
-					put_in_active_hand(I)
-					visible_message("<span class='warning'>[src] catches [I]!</span>")
-					throw_mode_off()
-					return
-
-		var/zone = ran_zone("chest", 65)
-		var/dtype = BRUTE
-		if(istype(I, /obj/item/weapon))
-			var/obj/item/weapon/W = I
-			dtype = W.damtype
-		var/throw_damage = I.throwforce*(speed/5)
-
-		I.throwing = 0		//it hit, so stop moving
-
-		if((I.thrower != src) && check_shields(throw_damage, "\the [I.name]", I, THROWN_PROJECTILE_ATTACK))
-			return
-
-		var/obj/item/organ/external/affecting = get_organ(zone)
-		if(!affecting)
-			var/missverb = (I.gender == PLURAL) ? "whizz" : "whizzes"
-			visible_message("<span class='notice'>\The [I] [missverb] past [src]'s missing [parse_zone(zone)]!</span>",
-				"<span class='notice'>\The [I] [missverb] past your missing [parse_zone(zone)]!</span>")
-			return
-		var/hit_area = affecting.name
-
-		src.visible_message("\red [src] has been hit in the [hit_area] by [I].")
-		var/armor = run_armor_check(affecting, "melee", "Your armor has protected your [hit_area].", "Your armor has softened hit to your [hit_area].", I.armour_penetration) //I guess "melee" is the best fit here
-
-		apply_damage(throw_damage, dtype, zone, armor, is_sharp(I), has_edge(I), I)
-
-		if(ismob(I.thrower))
-			var/mob/M = I.thrower
-			if(M)
-				add_logs(M, src, "hit", I, " (thrown)", print_attack_log = I.throwforce)
-
-		//thrown weapon embedded object code.
-		if(dtype == BRUTE && istype(I))
-			if(!I.is_robot_module())
+		I = AM
+		throwpower = I.throwforce
+		if(I.thrownby == src) //No throwing stuff at yourself to trigger reactions
+			return ..()
+	if(check_shields(throwpower, "\the [AM.name]", AM, THROWN_PROJECTILE_ATTACK))
+		hitpush = 0
+		skipcatch = 1
+		blocked = 1
+	/*else if(I)

```

Comment:
I'll allow this bit of commented code just to remind myself to port the object embedding stuff.